version: '3.8'
services:
  postgres:
    container_name: postgr
    image: postgres:15.1-alpine
    environment:
        POSTGRES_DB: "fastapi_database"
        POSTGRES_PASSWORD: "password"
    ports:
      - "5433:5432"
    networks:
      - api_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d fastapi_database -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 10
  ylab:
    container_name: ylab
    build:
      context: .
    command: "uvicorn main:app --reload --host 0.0.0.0"
    ports:
      - "8000:8000"
    volumes:
      - .:/app
    networks:
      - api_network
    depends_on:
      postgres:
        condition: service_healthy
  redis:
    container_name: redis
    image: redis:latest
    ports:
      - "6379:6379"
    networks:
      - api_network
    healthcheck:
      test: ["CMD-SHELL", "redis-cli", "ping"]
      interval: 5s
      timeout: 5s
      retries: 10
    depends_on:
      postgres:
        condition: service_healthy
  rabbit:
    container_name: rabbit
    image: rabbitmq:3.10.7-management
    ports:
      - "15672:15672"
      - "5672:5672"
    networks:
      - api_network
  celery_worker:
    container_name: celery_worker
    build:
      context: .
    command: celery -A config.cel_config.celery_app worker --loglevel=INFO -Q test-queue
    networks:
      - api_network
    depends_on:
      postgres:
        condition: service_healthy

networks:
  api_network:
    name: api_network
